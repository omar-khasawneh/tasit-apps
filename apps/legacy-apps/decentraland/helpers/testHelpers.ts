import { ESTATE, PARCEL } from "../constants/AssetTypes";
import { DONE, PENDING, MISSING } from "../constants/ActionStatus";
import { SUCCESSFUL } from "../constants/UserActionStatus";

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateWithoutName = {
  type: ESTATE,
  id: "321",
  name: "",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: "0",
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: "0",
  priceUSD: 0,
  asset: parcel,
};

function getIdAsync(): Promise<string> {
  return new Promise((resolve, reject) => {
    process.nextTick(() =>
      resolve("0x1234567890123456789012345678901234567890")
    );
  });
}

export const anAction = {
  getId: async (): Promise<string> => {
    return await getIdAsync();
  },
};

export const accountCreationSteps = [
  {
    name: "Account created",
    status: DONE,
  },
  {
    name: "ETH",
    status: PENDING,
  },
  {
    name: "MANA",
    status: MISSING,
  },
];

export const parcelUserAction = {
  ["0x1234567890123456789012345678901234567890"]: {
    assetId: "0123456789",
    status: SUCCESSFUL,
  },
};

export const estateUserAction = {
  ["0x987654321098765432109876543210987654321"]: {
    assetId: "123",
    status: SUCCESSFUL,
  },
};

export const userActions = { ...parcelUserAction, ...estateUserAction };

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
  anAction,
  estateWithoutName,
  parcelUserAction,
  estateUserAction,
  userActions,
};
