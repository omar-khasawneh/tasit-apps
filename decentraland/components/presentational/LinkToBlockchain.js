import React from "react";
import PropTypes from "prop-types";
import { Alert } from "react-native";
import { Button, Icon } from "native-base";
import {
  showError,
  openURL,
  buildBlockchainUrlFromAction,
  getNetworkName,
} from "@helpers";

const _openLinkOf = async action => {
  const url = await buildBlockchainUrlFromAction(action);
  try {
    await openURL(url);
  } catch (err) {
    showError(`Unable to open action link`);
  }
};

const _openLinkInfo = () => {
  const title = "";
  const message = `Shows the action details on the real chains.`;
  const buttons = [{ text: "Okay" }];
  Alert.alert(title, message, buttons);
};

const _onPress = action => {
  const supportedNetworks = ["ropsten"];
  const networkName = getNetworkName();
  const isNetworkSupported = supportedNetworks.includes(networkName);

  if (isNetworkSupported) _openLinkOf(action);
  else _openLinkInfo();
};

export default function LinkToBlockchain(props) {
  const { action } = props;
  if (!action) return null;

  return (
    <Button transparent onPress={_onPress}>
      <Icon name="eye" />
    </Button>
  );
}

LinkToBlockchain.propTypes = {
  action: PropTypes.object,
};
