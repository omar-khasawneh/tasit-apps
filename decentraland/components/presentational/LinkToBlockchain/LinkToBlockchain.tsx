import React from "react";

import { StyleSheet, Alert, TouchableOpacity } from "react-native";
import { Icon } from "native-base";
import {
  showError,
  openURL,
  buildBlockchainUrlFromActionId,
  getNetworkName,
} from "../../../helpers";
import Colors from "../../../constants/Colors";

const _openLinkOf = async actionId => {
  const url = buildBlockchainUrlFromActionId(actionId);
  try {
    await openURL(url);
  } catch (err) {
    showError(`Unable to open action link`);
  }
};

const _openLinkInfo = () => {
  const title = "";
  const message = `Shows the action details on the real chains.`;
  const buttons = [{ text: "Okay" }];
  Alert.alert(title, message, buttons);
};

const _onPress = actionId => {
  const supportedNetworks = ["ropsten"];
  const networkName = getNetworkName();
  const isNetworkSupported = supportedNetworks.includes(networkName);

  if (isNetworkSupported) _openLinkOf(actionId);
  else _openLinkInfo();
};

const styles = StyleSheet.create({
  icon: {
    color: Colors.linkColor,
  },
  touchable: {
    flexDirection: "column",
    justifyContent: "center",
  },
});

interface LinkToBlockchainProps {
  actionId: string;
}

const LinkToBlockchain: React.SFC<LinkToBlockchainProps> = ({ actionId }) => {
  if (!actionId) return null;

  const onPress = () => {
    _onPress(actionId);
  };

  return (
    <TouchableOpacity onPress={onPress} style={styles.touchable}>
      <Icon name="eye" style={styles.icon} />
    </TouchableOpacity>
  );
};

export default LinkToBlockchain;
