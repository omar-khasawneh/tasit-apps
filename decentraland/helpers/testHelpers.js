import AssetTypes from "@constants/AssetTypes";
import { DONE, PENDING, MISSING } from "@constants/ActionStatus";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateWithoutName = {
  type: ESTATE,
  id: "321",
  name: "",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: "0",
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: "0",
  priceUSD: 0,
  asset: parcel,
};

export const anAction = {
  getId: async () => "0x1234567890123456789012345678901234567890",
};

export const accountCreationSteps = [
  {
    name: "Account created",
    status: DONE,
  },
  {
    name: "ETH",
    status: PENDING,
  },
  {
    name: "MANA",
    status: MISSING,
  },
];

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
  anAction,
  estateWithoutName,
  accountCreationSteps,
};
