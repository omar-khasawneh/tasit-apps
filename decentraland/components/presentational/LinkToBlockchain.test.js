import React from "react";
import { shallow } from "enzyme";
import LinkToBlockchain from "./LinkToBlockchain";
import { anTempAction } from "@helpers/testHelpers";

describe("LinkToBlockchain", () => {
  describe("renders the component", () => {
    it("null action", async () => {
      const actionId = null;
      expect(
        shallow(<LinkToBlockchain actionId={actionId} />)
      ).toMatchSnapshot();
    });

    it("valid action", async () => {
      const actionId = await anTempAction.getId();
      expect(
        shallow(<LinkToBlockchain actionId={actionId} />)
      ).toMatchSnapshot();
    });
  });
});
