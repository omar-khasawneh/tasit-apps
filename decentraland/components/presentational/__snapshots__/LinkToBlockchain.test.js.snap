// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkToBlockchain renders the component null action 1`] = `""`;

exports[`LinkToBlockchain renders the component valid action 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Styled(Icon)
    name="eye"
    primary={true}
    style={
      Object {
        "color": "#2196f3",
      }
    }
  />
</TouchableOpacity>
`;
